### ISSUE TITLE
404 not found on accessing root (index.html)

### ISSUE THREAD

**Issue by robertwt7** [serverless-nextjs/serverless-next.js#522]
**Describe the bug**
Hey guys,

I'm gonna be honest with you, I just started playing with AWS for 2 months. Still very new to everything, so probably its just a dumb mistake (sorry!).

Came across this repo and tried to deploy 1 today. Really love the work!

I'm not sure if this is a bug that's been fixed, but I think I choose a stable version of your version (not the alpha one)

```
# serverless.yml
myApp:
  component: "@sls-next/serverless-component@1.15.1"
```

**To Reproduce**
All I did was serverless deploy, and boom when I go to my cloudfront link, it just shows 404 page not found.

Checked the lambda for routing and it looks fine, I'm just creating a simple few pages app. But the `static-pages/index.html` is there in s3 but doesn't seem to route there.

`api/api1.js` works though. So i'm not sure what's happening with my pages. Any help would be appreciated

Cheers guys

**Comment by robertwt7** [comment]
I assume that from this code:
```
const router = (manifest) => {
    const { pages: { ssr, html } } = manifest;
    const allDynamicRoutes = Object.assign(Object.assign({}, ssr.dynamic), html.dynamic);
    return (uri) => {
        let normalisedUri = uri;
        if (isDataRequest(uri)) {
            normalisedUri = uri
                .replace(`/_next/data/${manifest.buildId}`, "")
                .replace(".json", "");
        }
        if (ssr.nonDynamic[normalisedUri]) {
            return ssr.nonDynamic[normalisedUri];
        }
        console.log(uri);
        for (const route in allDynamicRoutes) {
            const { file, regex } = allDynamicRoutes[route];
            const re = new RegExp(regex, "i");
            const pathMatchesRoute = re.test(normalisedUri);
            if (pathMatchesRoute) {
                return file;
            }
        }
        if (html.nonDynamic["/404"] !== undefined) {
            return "pages/404.html";
        }
        return "pages/_error.js";
    };
};
```

It always returns `pages/404.html`, however I'm not really sure how to debug with this. Even after i did `console.log` every uri and redeploy to cloudFront, it doesn't go in CloudWatch...

**Comment by fredericocurti** [comment]
I'm having the exact same issue. If it helps figuring it out, I've added serverless-next.js to a app that didn't had the _app.tsx file which worked fine, but to my other app with _app.tsx i'm also getting 404s at index

**Comment by robertwt7** [comment]
Huh.. its the same, my app has `_app.js` as well. But i'm not sure if that's the issue, I thought `normaliseURI` is replacing `/` with `/index` which doesn't exist in my manifest. However changing the lambda to not use `normaliseURI` still gives me 404

**Comment by robertwt7** [comment]
![image](https://user-images.githubusercontent.com/15647967/88668123-8eeb4580-d125-11ea-98f0-c639422203f8.png)

Okay.. I just realise that the lambda does not invoke if i'm just accessing the domain straightaway. It only invokes whenever I put a custom route `/abc`, `/test`, `/index`. It doesn't get called if I don't put anything from the domain

`mydomain.com` -> doesn't trigger lambda

**Comment by fredericocurti** [comment]
Can you try deploying it again? I've cleared my AWS resources with aws-nuke and after deploying it again today it's back to working fine for some reason.

**Comment by zoozalp** [comment]
I'm having the same problem. Once it's in 404 page, links to / works just fine. Unfortunately I can't risk using aws-nuke right now. Problem started after upgrading nextjs to 9.5 last last night.

**Comment by robertwt7** [comment]
Yeah I might try to clear it up tonight and re-deploy. 

My nextjs version is  `"next": "v9.4.5-canary.32"`, what's yours @fredericocurti ?

**Comment by lone-cloud** [comment]
serverless-next.js needs to be updated to work with Next.js 9.5, but it sounds like you're on the right path with your fix.
I rewrote this serverless-next.js the way I think it should have been done in a separate library (can find from my profile). I very recently fixed this exact issue.

**Comment by fredericocurti** [comment]
@robertwt7 I downgraded manually to 9.4.4 and that seems to be working fine for now

**Comment by robertwt7** [comment]
Ah i got it now

After downgrading to next `v9.4.4` I realise that the index.html page is moved to `public/index.html` instead of `static-pages/index.html`

Thus this line won't work. Still doesn't answer why root url doesn't get logged though. Just went pass through it straightaway

```
    if (isHTMLPage || isPublicFile) {
        s3Origin.path = isHTMLPage ? "/static-pages" : "/public";
        addS3HostHeader(request, normalisedS3DomainName);
        if (isHTMLPage) {
            request.uri = `${uri}.html`;
        }
        return request;
    }
```

**Comment by danielcondemarin** [comment]
This should be fixed now in `@sls-next/serverless-component@1.16.0` 🎉

**Comment by zoozalp** [comment]
it works! thank you 💯
